### ISSUE TITLE
Support Classes without getters

### ISSUE THREAD

**Issue by mrieser** [javalin/javalin-openapi#63]
I just tried upgrading from Javalin w/ OpenAPI 4.x to 5.1.1.

For DTOs, we have lots of classes which define just public fields, like this:

```java
public class EntityDTO {
    public long id;
    public String key;
    public int otherValue;
}

// default empty constructor plus an all-fields-constructor omitted for this example
```

If we use such a type as a body content or response, it is reported as an empty json object (`{}`).

If we add getters for the fields (e.g. `public long getId() { return this.id; }`), then the fields start showing up in the API specification.

Is there any way to configure the OpenAPI-Processor to look at available fields, and not only at getters?

Alternatively, we would probably switch to records, if they would be supported (see #62). But adding lots of getters just to get the API documentation right feels somehow wrong as it just bloats up the code.

**Comment by dzikoysk** [comment]
In general this was conscious decision to use only getters, because that's the base of standardized dto object. Public properties are against encapsulation, not only in Java, but any code generator (Kotlin, POJO utilities, records, lombok etc) expose properties through dedicated accessors. It also at some point protects scheme generator to not pick random fields (because it'd would be an exposure of internal implementation).

Records would be natural upgrade for this kind of usages, alternatively we would need to implement some kind of opt-in support through a new annotation like @OpenApiByFields(visibility: by default PUBLIC)

**Comment by mrieser** [comment]
Thanks for the feedback, I see. In the past, I was against getters/setters for simple "record-like" objects as DTOs usually are that don't need complex logic, but just need to store some data.

Given it was a conscious decision feel free to close this issue with a "won't fix" and I'll keep my eyes on the support of Java records (#62).

Maybe a "Upgrade documentation" or a list of "incompatibilities with earlier OpenAPI support" as documentation would be nice. I understand that the old (reflection-based) approach had its problems and that this is now completely new code. Still, as it is kind of advertised as a successor, it would be great to know about such differences in behavior when attempting an upgrade.

Thanks a lot for your great work!

**Comment by dzikoysk** [comment]
Even if it was intentional approach to enforce properties, I'm not against providing alternative tool to cover some extra cases like fields. I'll take a look on sth like ~`@OpenApiByFields` during records (#62), so let's keep it open. I won't be able to play with it today, but I'll try to check it this week.

**Comment by dzikoysk** [comment]
Records:

https://github.com/javalin/javalin-openapi/blob/3e26ab1f4b54a720526e5cb59a6da6a027df25ec/examples/javalin-gradle-kotlin/src/main/java/io/javalin/openapi/plugin/test/JavalinTest.java#L329-L330

Fields:

https://github.com/javalin/javalin-openapi/blob/3e26ab1f4b54a720526e5cb59a6da6a027df25ec/examples/javalin-gradle-kotlin/src/main/java/io/javalin/openapi/plugin/test/JavalinTest.java#L332-L339

Both features are available in the latest 5.1.2-SNAPSHOT :)

**Comment by mrieser** [comment]
Thank you very much for this fast response. I'll plan to give a try very soon!
